internal/pkg/deploy/pipeline.go

```diff
@@ -159,7 +159,14 @@ func (s *Source) Owner() (string, error) {
 // is deloying to and the containerized applications that will be deployed.
 type PipelineStage struct {
 	*AssociatedEnvironment
-	LocalApplications []string
+	LocalApplications []AppInStage
```

Comment:
Guess it's kinda like half and half. In general, using pointer is beneficial when your struct is large and you want to avoid copying. However, in this case the struct is very small so it does not really matter that much